docker/code-check.Dockerfile

```diff
@@ -0,0 +1,7 @@
+# Image for all tasks related to static code analysis in Async-Profiler
+FROM public.ecr.aws/docker/library/amazoncorretto:11-alpine-jdk
+
+ADD https://raw.githubusercontent.com/llvm/llvm-project/refs/tags/llvmorg-20.1.7/clang-tools-extra/clang-tidy/tool/clang-tidy-diff.py /usr/bin/
+RUN apk add --no-cache clang-extra-tools linux-headers make python3 git && \
+    chmod +x /usr/bin/clang-tidy-diff.py
```

Comment:
I see `ADD` has [`--chmod`](https://docs.docker.com/reference/dockerfile/#add---chown---chmod) option, does it work here?

Reply:
Right: 32d9bb3b12ce8af463d50789bec9fba3d6467cbb